dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/metadata/ConfigurableMetadataServiceExporter.java

```diff
@@ -146,6 +145,9 @@ private ProtocolConfig generateMetadataProtocol() {
         } else {
             protocolConfig.setPort(port);
         }
+        
+        applicationModel.getApplicationConfigManager().getProtocol(specifiedProtocol)
+            .ifPresent(protocolConfig::mergeProtocol);
```

Comment:
Will this override `port` configuration?

Reply:
> Will this override `port` configuration?

No, it will only work on variables with null values